app/views/recipes/show.html.erb

```diff
@@ -1 +1,58 @@
-<h2>Recipe Details Page</h2>
\ No newline at end of file
+<section class="show_recipe">
+
+  <%= link_to 'Back to Recipes', recipes_path, class: 'btn' %>
+
+  <article class="upper_section">
+    <h2 class="recipe_heading"><%=@recipe.name %></h2>
+
+    <div class="recipe_details">
+      <div class="preparation_public">
+        <p class="preparation_time"><%= "Preparation Time: #{@recipe.preparation_time} hour"%></p>
+        <% if !current_user.nil? && current_user.id == @recipe.user_id %>
+          <%= form_for @recipe do |f| %>
+            <div class="public_btn_container">
+              <%= f.label :public, class: "visibility_toggle_lable", for: "visibility-toggle" %>
+              <%= f.check_box :public, id: "visibility-toggle", class: "visibility_toggle_input" %>
+            </div>
+
+            <%= f.submit "Update Visibility", class: "btn" %>
+          <%end%>
+        <% end %>
+      </div>
+      <p><%= "Cooking Time: #{@recipe.cooking_time} hour"%></p>
+      <p><%= "Steps to prepare: #{@recipe.description}"%></p>
+    </div>
+  </article>
+  <% if !current_user.nil? && current_user.id == @recipe.user_id %>
+    <div class="buttons">
+      <%= link_to 'Generate Shopping List', recipe_shopping_list_path(@recipe.id), class: 'btn' %>
+      <%= link_to 'Add Ingredient', new_recipe_recipe_food_path(@recipe.id), class: 'btn' %>
+    </div>
```

Comment:
### [REQUIRED] Please fix the NoMethodError
![image](https://github.com/MNisarAli/recipe_app/assets/24216198/e18e642a-3883-4f97-bf10-e54024dfc70d)

Reply:
Hi @anita00001, all the actions and buttons in the recipe details page are referencing ‘food’ and ‘shoping list’ and the errors can only be fixed with implementation of these models, which is not part of this project. Moreover when we finish working on all the project requirements it will make a link to that stuff automatically.

Reply:
### Hi @MNisarAli , 
Yes, you are right!
I hope you won't mind removing/commenting on lines that cannot be executed at the moment so that it does not throw an Error when rendered.

### Please feel free to contact me if you need more help.

### Best Regards,
Anita Sharma
@anita00001